### DISCUSSION TITLE
Serving Mastodon over http

### DISCUSSION THREAD

**Discussion by cerebrate** [mastodon/mastodon#21421] [Q&A] [answered]
Not to the internet, you understand, but I have an nginx reverse proxy at the edge of my network already that applies the https certificates to services that I publish externally, and I'd prefer not to be bothered having to deal with certificates for purely internal hosts if I don't absolutely have to.

Possible? (My first attempt at this ended up at a 301 loop.)

Thanks in advance.

**Comment by mfenniak** [comment, marked as answer]
You likely didn't configure the `X-Forwarded-Proto` HTTP header on your nginx proxy.  This header allows Mastodon to recognize that it's being served under https despite the direct connection not being https; therefore it would issue a redirect to https.

A typical nginx config would have:
https://github.com/mastodon/mastodon/blob/53817294fc95eabfed6129138f9aaa920e13c4b9/dist/nginx.conf#L116-L120
Among other proxy settings that you might want to review.

**Reply by cerebrate** [reply]
Yep, looks like that was it. Thanks!
